homeassistant/components/shelly/sensor.py

```diff
@@ -989,6 +989,24 @@ class RestSensorDescription(RestEntityDescription, SensorEntityDescription):
             or status[key]["counts"].get("xtotal") is None
         ),
     ),
+    "counter_frequency": RpcSensorDescription(
+        key="input",
+        sub_key="counts",
+        name="Pulse counter frequency",
+        native_unit_of_measurement=UnitOfFrequency.HERTZ,
+        state_class=SensorStateClass.MEASUREMENT,
+        value=lambda status, _: status["freq"],
+        removal_condition=lambda config, status, key: (config[key]["enable"] is False),
+    ),
+    "counter_frequency_value": RpcSensorDescription(
+        key="input",
+        sub_key="counts",
+        name="Pulse counter frequency value",
```

Comment:
The unit and formula for calculating value are modifiable by the user during runtime, so we do not add a unit and state class for this sensor.